api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -318,13 +316,6 @@ private void validateUpdatedProfile(Profile updatedProfile, Profile prevProfile)
       // See RW-1488.
       throw new BadRequestException("Changing username is not supported");
     }
-    final VerifiedInstitutionalAffiliation updatedAffil =
```

Comment:
We're not changing the fact that non-admin profile updates aren't allowed to change affiliation, right? Removing this line breaks that enforcement.

I think I know why you removed this, because we're using a unified code path for non-admin and admin profile updates. But we need to account for the fact that self-profile updates have different business logic than admin-driven profile updates.

A couple options:
 - Bump this logic into a `checkUserProfileUpdate` method, whose job is basically to reject any requests where a user is trying to update things that they shouldn't be allowed to. This fits nicely with a higher-level goal of pushing more of the core upsert logic into a service, but keeping the top-level "can I do this" logic in the controller itself.
 - Pass down some additional call context to methods like this, e.g. "boolean isAdmin" (Seems clunky)

Reply:
We can't control the signature of the public controller methods without changing the API and that seems like overkill for something like this. so I'll do the first one.

Reply:
hm. I don't think we have a pattern of checking authority on the backend outside of `@AuthorityRequired` annotations and I don't want to introduce one. 🙁 maybe I'll fork the controller now and factor out as much stuff from updateProfile as I can.

Reply:
I don't think forking the controller should be needed. The crux of this issue (similar to the one further down about `userProvider.get()`) is that we're attempting to use the same top-level controller method for two different purposes now.

Instead, one could create a private `updateProfileImpl` method which is called by the two top-level controller methods. With that change in place, to resolve this issue you could create a `checkUserProfileUpdate` method which is only called by the `updateProfile` top-level controller.

e.g. something like the following would be very clear

```
public ResponseEntity<Void> updateProfile(Profile updatedProfile) {
  validateUserProfileUpdate(updatedProfile);
  updateProfileImpl(updatedProfile, userProvider.get());
}

public ResponseEntity<Void> updateVerifiedInstitutionalAffiliation(long userId, VerifiedInstitutionalAffiliation verifiedInstitution) {
  [...]
  updateProfileImpl(updatedProfile, userDao.findUserByUserId(userId));
}
```

and eventually updateProfileImpl should be factored into ProfileService.

Reply:
moved into updatedProfile.